src/vivarium_public_health/risks/effect.py

```diff
@@ -316,3 +320,158 @@ def _get_risk_exposure_class(self, builder: Builder) -> Risk:
                 f"Risk effect model {self.name} requires a Risk component named {self.risk}"
             )
         return risk_exposure_component
+
+
+class NonLogLinearRiskEffect(RiskEffect):
+    """Risk effect for exposure-parametrized relative risks."""
+
+    ##############
+    # Properties #
+    ##############
+
+    @property
+    def configuration_defaults(self) -> Dict[str, Any]:
+        """
+        A dictionary containing the defaults for any configurations managed by
+        this component.
+        """
+        return {
+            self.name: {
+                "data_sources": {
+                    "relative_risk": f"{self.risk}.relative_risk",
+                    "population_attributable_fraction": f"{self.risk}.population_attributable_fraction",
+                },
+            }
+        }
+
+    @property
+    def columns_created(self) -> List[str]:
+        return [f"{self.risk.name}_exposure"]
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        exposure_col = pd.DataFrame(columns=self.columns_created)
+        self.population_view.update(exposure_col)
+
+    def on_time_step_prepare(self, event: Event) -> None:
+        # exposure_values = self.exposure(event.index)
+        # TODO: temporary workaround, investigate why exposure is always 0
```

Comment:
I'm going to make a guess - exposure values are 0 because of some interaction that happens with subclassing RiskEffect. I'm curious what functionality you got from subclassing that component since you still had to write a lot of code and if it would just be worth inheriting from component instead.

Reply:
FWIW I think you should subclass from RiskEffet but if this component is so different that you don't gain any value from it then maybe you don't need to

Reply:
I've taken a look and I think there would be enough copy and paste that it's worth subclassing.